src/main/java/com/monumental/services/GoogleMapsService.java

```diff
@@ -64,6 +69,44 @@ public Geometry getCoordinatesFromAddress(String address) {
         return null;
     }
 
+    private AddressBundle buildBundle(GeocodingResult result) {
+        AddressBundle bundle = new AddressBundle();
+        String country = null;
+        for (AddressComponent component : result.addressComponents) {
+            for (AddressComponentType type : component.types) {
+                switch (type) {
+                    case LOCALITY:
+                        bundle.city = component.longName;
+                        break;
+                    case ADMINISTRATIVE_AREA_LEVEL_1:
+                        bundle.state = component.shortName;
+                        break;
+                    case COUNTRY:
+                        country = component.shortName;
+                        break;
+                }
+            }
+        }
+        /* US Territories are usually listed as countries, with some strange edge cases, like
+           San Juan, Puerto Rico where the administrative_area_level_1 is San Juan, not Puerto Rico.
+           For Guam, there is no administrative_area_level_1, and Guam is the country
+           This is kind of a catch all attempt to get a 2 letter state code
+         */
+        if ((bundle.state == null || bundle.state.length() != 2) && country != null && country.length() == 2) {
+            bundle.state = country;
+        }
+        /* If there was no 2 letter state code, we could end up with something longer like San Juan, so fallback to
+           leaving the field blank if it's not 2 letters long
+         */
+        if (bundle.state == null || bundle.state.length() != 2) {
```

Comment:
I think the first part of this `if` is redundant

Reply:
It should probably be `bundle.state != null && bundle.state.length() != 2`

Reply:
It's checking if we didn't find a state, or if we did and it's not two letters long, which can happen for places like San Juan, Puerto Rico where the short_name of the state is actually `San Juan`

Reply:
I know, but we're checking if `bundle.state` is `null` and if it is, we set `bundle.state` equal to `null`. Am I missing something?

Reply:
You can't check `bundle.state.length` without checking if `bundle.state` exists, otherwise you'll get an exception. I suppose `||` or `&&` both work here depending on `==` or `!=`

Reply:
I think `&&` makes more sense when you read it